src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -138,13 +138,16 @@ protected function processReturn(File $phpcsFile, $stackPtr, $commentStart)
                             // exiting, then there is no problem.
                             $semicolon = $phpcsFile->findNext(T_WHITESPACE, ($returnToken + 1), null, true);
                             if ($tokens[$semicolon]['code'] !== T_SEMICOLON) {
-                                $error = 'Function return type is void, but function contains return statement';
+                                $error = "Function return type is {$returnType}, but function contains return statement";
                                 $phpcsFile->addError($error, $return, 'InvalidReturnVoid');
                             }
                         }
                     }//end if
-                } else if ($returnType !== 'mixed' && in_array('void', $typeNames, true) === false) {
-                    // If return type is not void, there needs to be a return statement
+                } else if ($returnType !== 'mixed'
+                    && in_array('never', $typeNames) === false
+                    && in_array('void', $typeNames) === false
```

Comment:
Yeah, I agree. The name of `never` says it all; the function never returns. So it couldn't be part of a union because that would mean it _does_ return something at least sometimes.

Similarly, I think `void` in a union also never makes sense (and should be replaced with `null`), but in the context of this PR I think it should be left as is.

Reply:
> Similarly, I think `void` in a union also never makes sense (and should be replaced with `null`), but in the context of this PR I think it should be left as is.

Agreed.